DFC.Digital/DFC.Digital.Web.Sitefinity.DfcSearchModule/SearchIndexEnhancers/JobProfileIndexEnhancer.cs

```diff
@@ -25,49 +29,60 @@ public JobProfileIndexEnhancer(IJobProfileRepository jobProfileRepository, IJobP
 
         public JobProfile JobProfile { get; private set; }
 
-        public void Initialise(JobProfileIndex jobProfileIndex)
+        public void Initialise(JobProfileIndex jobProfileIndex, bool isPublishing)
         {
-           this.JobProfile = jobProfileRepository.GetByUrlName(jobProfileIndex.UrlName);
+            if (jobProfileIndex == null)
+            {
+                throw new ArgumentNullException(nameof(jobProfileIndex));
+            }
+
+            this.JobProfile = isPublishing ? jobProfileRepository.GetByUrlNameForSearchIndex(jobProfileIndex.UrlName) : jobProfileRepository.GetByUrlName(jobProfileIndex.UrlName);
+            this.jobProfileIndex = jobProfileIndex;
         }
 
-        public JobProfileIndex GetRelatedFieldsWithUrl(JobProfileIndex jobProfileIndex)
+        public void PopulateRelatedFieldsWithUrl()
         {
             if (JobProfile != null)
             {
-                jobProfileIndex.JobProfileCategoriesWithUrl = GetJobProfileCategoriesWithUrl(JobProfile);
+                jobProfileIndex.JobProfileCategoriesWithUrl = GetJobProfileCategoriesWithUrl();
 
-                jobProfileIndex.Interests = JobProfile.RelatedInterests;
-                jobProfileIndex.Enablers = JobProfile.RelatedEnablers;
-                jobProfileIndex.EntryQualifications = JobProfile.RelatedEntryQualifications;
-                jobProfileIndex.TrainingRoutes = JobProfile.RelatedTrainingRoutes;
-                jobProfileIndex.JobAreas = JobProfile.RelatedJobAreas;
-                jobProfileIndex.PreferredTaskTypes = JobProfile.RelatedPreferredTaskTypes;
+                jobProfileIndex.Interests = JobProfile.RelatedInterests.ToList();
```

Comment:
These have been ok to-date with IQueryables, is this dues to the circuit breaker changes?

Reply:
No this was the bug that caused the search index to break on publish. Publishing pipeline follows a different path as opposed to the re-index, the publishing path container lifetime is valid only until the end of request, causing the IQueryable to lose its context to execute. Since we are assigning it to the index, it is as a good a place to execute the query and store the results.